test/test_pyfftw_numpy_interface.py

```diff
@@ -50,8 +50,20 @@
     from ._cook_nd_args import _cook_nd_args
     numpy.fft.fftpack._cook_nd_args = _cook_nd_args
 
-complex_dtypes = (numpy.complex64, numpy.complex64, numpy.complex128, numpy.clongdouble)
-real_dtypes = (numpy.float16, numpy.float32, numpy.float64, numpy.longdouble)
+complex_dtypes = []
+real_dtypes = []
+if '32' in _supported_types:
+    complex_dtypes.extend([numpy.complex64]*2)
+    real_dtypes.extend([numpy.float16, numpy.float32])
+if '64' in _supported_types:
+    complex_dtypes.append(numpy.complex128)
+    real_dtypes.append(numpy.float64)
+if 'ld' in _supported_types:
+    complex_dtypes.append(numpy.clongdouble)
+    real_dtypes.append(numpy.longdouble)
+
+# complex_dtypes = (numpy.complex64, numpy.complex64, numpy.complex128, numpy.clongdouble)
+# real_dtypes = (numpy.float16, numpy.float32, numpy.float64, numpy.longdouble)
```

Comment:
Nice.
Unless I missed it, you will also need to fix this in the scipy interfaces for versions < 0.19 because those currently don't import from these numpy ones.:

https://github.com/pyFFTW/pyFFTW/blob/60b81051e4a09a7fd599abd3123aaaa6fe02932d/test/test_pyfftw_scipy_interface.py#L82

You could import from numpy ones there and just drop the unsupported `float16', 'complex64' pair.  (float16 support in scipy.fftpack was fixed in 0.19)

Reply:
Yes, this is still work in progress. If all libraries are available as before, all tests pass. But if I build w/o single precision, there are still multiple tests I need to patch, scipy is one of them. I have to think of a good way of automated testing in that case. I may have to add environment variables that `setup.py` picks up to ignore for example single precision.